.github/workflows/build-plugin-archive.yml

```
@@ -0,0 +1,241 @@
+
+
+      #
+      # JS environment. Only for setting up node/npm versions. Actual tasks should be run at project level
+      #
+
+      - name: Check for package.json
+        id: check-for-package
+        run: |
+          if [ -f package.json ]; then
+            echo "has_package=true" >> $GITHUB_OUTPUT
+          else
+            echo "has_package=false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Set up node
+        if: steps.check-for-package.outputs.has_package == 'true'
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ inputs.NODE_VERSION }}
+          registry-url: ${{ inputs.NPM_REGISTRY_DOMAIN }}
+
+      - name: Cache npm dependencies
+        if: steps.check-for-package.outputs.has_package == 'true'
+        uses: actions/cache@v3
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      #
+      # Package metadata
+      #
+
+      - name: Add commit hash to plugin header
+        env:
+          GIT_SHA: ${{ steps.git.outputs.sha_short }}
+        run: 'sed -Ei "s/SHA: .*/SHA: ${GIT_SHA}/g" ${{ inputs.PLUGIN_MAIN_FILE }}'
+
+      - name: Set plugin version header
+        env:
+          PACKAGE_VERSION: ${{ github.event.inputs.packageVersion }}
+        run: 'sed -Ei "s/Version: .*/Version: ${PACKAGE_VERSION}/g" ${{ inputs.PLUGIN_MAIN_FILE }}'
+
+      #
+      # Archive tooling
+      #
+
+      - name: Install WP-CLI
+        run: |
+          mkdir ~/wp-cli && cd ~/wp-cli
+          curl -L https://raw.githubusercontent.com/wp-cli/builds/gh-pages/phar/wp-cli.phar --output wp
+          chmod +x wp
+          sudo ln -s ~/wp-cli/wp /usr/local/bin/wp
+
+      - name: Install dist-archive command
+        run: wp package install wp-cli/dist-archive-command
+
+      - name: Create dist-archive php parameters from inputs
+        id: dist-archive
+        run: |
+          PARAMS=""
+          FILE='${{ inputs.EXTRA_PHP_FILE }}'
+          CODE=$(cat <<'EOF'
+          ${{ inputs.EXTRA_PHP }}
+          EOF
+          )
+          if [ ! -z "$FILE" ]; then
+            PARAMS+="--require=$FILE "
+          fi
+          if [ ! -z "$CODE" ]; then
+            PARAMS+="--exec=${CODE@Q} "
+          fi
+          echo "params=${PARAMS}" >> $GITHUB_OUTPUT
+
+      #
+      # Archive/Artifact creation
+      #
+
+      - name: Run wp-cli command
+        run: wp dist-archive . ./archive.zip ${{ steps.dist-archive.outputs.params }} --plugin-dirname=${{ steps.plugin-data.outputs.plugin-name }}
+
+        # GitHub Action artifacts would otherwise produce a zip within a zip
+      - name: Unzip archive to dist/
+        run: unzip archive.zip -d dist
+
+      - id: set-artifact-name
+        run: echo "artifact=${{ steps.plugin-data.outputs.plugin-name }}-${{ inputs.PLUGIN_VERSION }}" >> $GITHUB_OUTPUT
+
+      - uses: actions/upload-artifact@v3
```

You used named commands previously, so let's continue this here.
```suggestion
      - name: Set artifact name
        id: set-artifact-name
        run: echo "artifact=${{ steps.plugin-data.outputs.plugin-name }}-${{ inputs.PLUGIN_VERSION }}" >> $GITHUB_OUTPUT


      - name: Upload artifact
        uses: actions/upload-artifact@v3
```